multi-transfer-esdt/src/lib.rs

```diff
@@ -66,26 +67,21 @@ pub trait MultiTransferEsdt:
                 continue;
             }
 
-            self.send()
-                .esdt_local_mint(&eth_tx.token_id, 0, &eth_tx.amount);
+            let _: IgnoreValue = self
+                .get_esdt_safe_contract_proxy_instance()
+                .mint_token(&eth_tx.token_id, &eth_tx.amount)
+                .with_esdt_transfer((eth_tx.token_id.clone(), 0, eth_tx.amount.clone()))
+                .execute_on_dest_context();
 
             // emit event before the actual transfer so we don't have to save the tx_nonces as well
             self.transfer_performed_event(batch_id, eth_tx.tx_nonce);
 
-            if self.blockchain().is_smart_contract(&eth_tx.to.clone()) {
-                let _: IgnoreValue = self
-                    .get_bridge_proxy_contract_proxy_instance()
-                    .deposit(&eth_tx)
-                    .with_esdt_transfer((eth_tx.token_id.clone(), 0, eth_tx.amount.clone()))
-                    .execute_on_dest_context();
-            } else {
-                valid_dest_addresses_list.push(eth_tx.to);
-                valid_payments_list.push(EsdtTokenPayment::new(eth_tx.token_id, 0, eth_tx.amount));
-            }
+            valid_tx_list.push(eth_tx.clone());
+            valid_payments_list.push(EsdtTokenPayment::new(eth_tx.token_id, 0, eth_tx.amount));
```

Comment:
For better consistency, I would update the `mintToken` endpoint to return the minted payment, and use that ESDTTokenPayment here instead of creating a new one.